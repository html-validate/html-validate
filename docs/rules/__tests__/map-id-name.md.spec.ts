import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<map name="foo" id="bar"></map>`;
markup["correct"] = `<map name="foo" id="foo"></map>`;

describe("docs/rules/map-id-name.md", () => {
	it("inline validation: incorrect", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"map-id-name":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"map-id-name":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
});
