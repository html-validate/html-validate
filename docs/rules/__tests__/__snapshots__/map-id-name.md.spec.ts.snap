// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`docs/rules/map-id-name.md inline validation: correct 1`] = `[]`;

exports[`docs/rules/map-id-name.md inline validation: incorrect 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 21,
        "line": 1,
        "message": ""id" and "name" attribute must be the same on <map> elements",
        "offset": 20,
        "ruleId": "map-id-name",
        "ruleUrl": "https://html-validate.org/rules/map-id-name.html",
        "selector": "#bar",
        "severity": 2,
        "size": 3,
      },
    ],
    "source": "<map name="foo" id="bar"></map>",
    "warningCount": 0,
  },
]
`;
